src/org/loklak/harvester/YoutubeScraper.java

```diff
@@ -205,9 +203,14 @@ public static JSONObject parseVideo(final BufferedReader br) throws IOException
         return json;
     }
     
-    private static long parseNumber(String n) throws NumberFormatException {
-        String withoutSpaces = CharMatcher.WHITESPACE.removeFrom(n);
-        return Long.parseLong(withoutSpaces);
+    private static String parseValue(String n) {
+        String number_value = CharMatcher.WHITESPACE.removeFrom(n);
+        Pattern number_format = Pattern.compile("((\\d)\\d{0,2})(,?\\d{3})*(\\.[\\d]+)?([MBK]?)");
```

Comment:
@vibhcool: The regex you mentioned is 
```
((\d)\d{0,2})(,?\d{3})*(\.[\d]+)?([MBK]?)
```
If we try to parse `123,45`, here's what happens - 
```
  1 -> \d
 23 -> \d{2}
,45 -> No match, fall back, check for acceptance of previous substring
123 -> passes as \d\d{2}
  , -> No match, skip
  4 -> \d
  5 -> \d{1}
EOF -> 45 passes as \d\d{1}
```

So, we have two matches for `123,45`. What went wrong? The fact that your regex tries to match any possible substring. A better solution would be
```
^((\d)\d{0,2})(,?\d{3})*(\.[\d]+)?([MBK]?)$
```
This will do things fine, but regex can still be a lot simpler - 
```
^\d{1,3}(,\d{3})*(\.\d+)?[MBK]$
```
Breaking it down,
```
        ^ - Start of string
  \d{1,3} - X or XX or XXX
(,\d{3})* - ,XXX repeated zero or more times
 (\.\d+)? - .X or .XX or .XXX ...
   [MBK]? - M or B or K occurs 0 or 1 times
        $ - End of string
```

regex101.com is a goog website.